consensus/replay.go

```diff
@@ -320,6 +326,7 @@ func (h *Handshaker) replayBlocks(proxyApp proxy.AppConns, appBlockHeight, store
 func (h *Handshaker) replayLastBlock(proxyApp proxy.AppConnConsensus) ([]byte, error) {
 	mempool := types.MockMempool{}
 	cs := NewConsensusState(h.config, h.state, proxyApp, h.store, mempool)
+	defer cs.Stop()
```

Comment:
Smells fishy to me...

If I look at stdlib, for example `file, err := os.Open(path)`.  If `err != nil` then I can just do error handling.  If `err == nil`, then the file is created, I can do something with it, and must remember to close it (or `defer file.Close()` in the line after error checking).  In this case, if I call `file.Close()` when an error is returned, I may trigger a panic.

Why not have our services do the same?  If they return an error, they failed, and *leave no state that needs cleanup* if they return success, then the caller must be responsible for stopping them and doing cleanup.  Is there a reason such assumptions would not work?